### ISSUE TITLE
Fix Critical A11Y Issues for extension console mods page

### ISSUE THREAD

**Issue by fungairino** [pixiebrix/pixiebrix-extension#7900]
We currently have 9 critical accessibility rules that are in violation in the extension console mods page as identified by
axe-core. These are:
- "color-contrast"
- "heading-order"
- "label-title-only"
- "landmark-one-main"
- "landmark-unique"
- "link-in-text-block"
- "list"
- "page-has-heading-one"
- "region"
https://github.com/dequelabs/axe-core/blob/develop/doc/rule-descriptions.md

## Additional context

As a larger effort of tackling a11y issues, this particular issue is just scoped to axe-core critical impact issues that
have been identified on the landing page for the extension console (the mods table page)

Axe-core describes critical issues the following way:
> Results in blocked content for people with disabilities, and will definitely prevent them from accessing fundamental features or content. This type of issue puts your organization at risk. Prioritize fixing as soon as possible, within the week if possible. Remediation should be a top priority.

https://github.com/dequelabs/axe-core/blob/develop/doc/issue_impact.md#critical

**Comment by fungairino** [comment]
Most of these should be straightforward fixes, with some UX coordination on colors for `color-contrast` and `link-in-text-block`.
